src/commands/Admin/eval.ts

```diff
@@ -20,35 +22,74 @@ export default class extends Command {
-		try {
-			const code: string = args.join(" ");
-			let evaled = eval(code);
+		const start: number = new Date().getTime();
+		const code: string = args.join(" ");
+		let evaled: any;
+		let output: any;
+		let isFile: boolean = false;
 
-			if (typeof evaled !== "string") evaled = require("util").inspect(evaled);
-			if (evaled.includes(process.env.TOKEN)) evaled = evaled.replace(process.env.TOKEN, "Y0U.TH0UGHT.1_W0ULD.L34V3_MY_D1SC0RD_T0K3N_H3R3");
-			this.client.log.info(`[DEVELOPER] ${context.author.tag} (${context.author.id}) ran eval with the code: ${code}`);
+		this.client.log.info(`[DEVELOPER] ${context.author.tag} (${context.author.id}) ran eval with the code: ${code}`);
 
-			if (evaled.length > 2000) {
-				writeFile(`${__dirname}/../../../files/EVAL-${context.guild?.id}.js`, await clean(evaled), function (err) {
-					if (err) console.error(err);
-				});
+		let description: string = `**Input**\n\`\`\`js\n${code}\n\`\`\``;
+
+		try {
+			if (code.includes("token")) this.client.token = `${Buffer.from(this.client.user!?.id).toString("base64")}.${genString(7)}.${genString(27)}`;
+
+			evaled = await eval(code);
+			description += `\n**Type:** ${typeof evaled}`;
+			if (typeof evaled !== "string") evaled = inspect(evaled);
 
-				await context.channel.send({
-					content: "The evaled code is more than 2000 characters so I am giving you a file instead, have fun 🙂",
-					files: [`${__dirname}/../../../files/EVAL-${context.guild?.id}.js`],
+			// @ts-ignore, will this break anything prob not idk :shrug:
+			if (code.includes("token")) this.client.token = process.env.TOKEN;
+			if (evaled.length < 1950) output = `**Output**\n\`\`\`js\n${evaled}\n\`\`\``;
+			else {
+				writeFile(`${__dirname}/../../../files/EVAL-${context.guild?.id}.js`, evaled, (err: any) => {
+					if (err) this.client.log.error(err);
 				});
 
-				return;
+				isFile = true;
 			}
 
-			await context.channel.send(`\`\`\`js\n${evaled}\n\`\`\``);
+			embed.setColor("GREEN");
 		} catch (err: any) {
-			await context.channel.send(`\`ERROR\` \`\`\`xl\n${await clean(err)}\n\`\`\``);
+			description += `\n**Error:** ${err.name}\n> ${err.message}\n \`\`\`\n${err.stack}\n\`\`\``;
+			Math.floor(Math.random() * 10) === 0 ? embed.setImage("https://i.imgur.com/YHb5kNm.gif") : null;
+			embed.setColor("RED");
 		}
+
+		const end: number = new Date().getTime();
+
+		embed.setDescription(description);
+		embed.setAuthor(`Run time: ${start - end} ms`);
+
+		context.reply({
+			embeds: [embed],
+			content: output,
+			files: isFile
+				? [
+						{
+							attachment: `${__dirname}/../../../files/EVAL-${context.guild?.id}.js`,
+							name: "eval.js",
+						},
+				  ]
+				: [],
+		});
 	}
 }
+
+function genString(l: number) {
+	var res = "";
+	const chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+	for (var i = 0; i < l; i++) res += chars.charAt(Math.floor(Math.random() * chars.length));
+
+	return Buffer.from(res).toString("base64").substring(0, l);
```

Comment:
btoa is deprecated

Reply:
What says that? It doesn't say that in the [living web APIs specification](https://html.spec.whatwg.org/multipage/webappapis.html#dom-btoa-dev)

Reply:
Decodes a string into bytes using Latin-1 (ISO-8859), and encodes those bytes into a string using Base64.
The data may be any JavaScript-value that can be coerced into a string.
This function is only provided for compatibility with legacy web platform APIs and should never be used in new code, because they use strings to represent binary data and predate the introduction of typed arrays in JavaScript. For code running using Node.js APIs, converting between base64-encoded strings and binary data should be performed using Buffer.from(str, 'base64') andbuf.toString('base64').

@since — v15.13.0
@deprecated — Use buf.toString('base64') instead.
@param data — An ASCII (Latin1) string.

Reply:
Our usecase isn't actually encoding binary data (which typed arrays should be used for). But since it's deprecated within Node I'm fine with sticking to Buffer instead if you want.